### PR TITLE
okhttp: forceful close after MAX_CONNECTION_AGE_GRACE_TIME

### PR DESCRIPTION
**Pull request by YifeiZhuang**
fixes https://github.com/grpc/grpc-java/issues/9721

### CODE REVIEW

File: okhttp/src/main/java/io/grpc/okhttp/OkHttpServerTransport.java [hunk 1]
```diff
@@ -926,7 +935,7 @@ public void ping(boolean ack, int payload1, int payload2) {
           return;
         }
         if (GRACEFUL_SHUTDOWN_PING == payload) {
-          triggerGracefulSecondGoaway();
+          triggerGracefulSecondGoaway(null);
```
[the remark this answers is not in the document]
Comment (YifeiZhuang) [reply okhttp/src/main/java/io/grpc/okhttp/OkHttpServerTransport.java]: Ah right that's true.
So after the first goaway and ping, the secondGoawayTimer is set with 1s ping timeout.
A normal successful ping ack within 1s cancels the timer and trigger second goAway and graceful shutdown. No ping after 1s also triggers the same path.

The ping method does not take any more parameter, and it implements frame handler interface. So I made the grace time a field. And it is synchronized on the lock to allow read and write to be thread safe.

[the remark this answers is not in the document]
Comment (YifeiZhuang) [reply okhttp/src/main/java/io/grpc/okhttp/OkHttpServerTransport.java]: @ejona86 wdyt?

File: okhttp/src/main/java/io/grpc/okhttp/OkHttpServerTransport.java [hunk 2]
```diff
@@ -289,6 +294,10 @@ private void triggerGracefulSecondGoaway() {
       } else {
         frameWriter.flush();
       }
+      if (gracefulShutdownPeriod != null) {
+        forcefulCloseTimer = scheduledExecutorService.schedule(
+            this::triggerForcefulClose, gracefulShutdownPeriod, TimeUnit.NANOSECONDS);
```
Comment (ejona86) [inline okhttp/src/main/java/io/grpc/okhttp/OkHttpServerTransport.java:299]: This changes the behavior when someone is doing:

```java
server.shutdown();
server.awaitTerminated(10000, TimeUnit.SECONDS);
```

Comment (YifeiZhuang) [reply okhttp/src/main/java/io/grpc/okhttp/OkHttpServerTransport.java:299]: I forgot to mention that https://github.com/grpc/grpc-java/pull/9649 changed the behavior of this. So I put it back.
So sound like we should do graceful shutdown when `server.shutdown`?

Comment (YifeiZhuang) [reply okhttp/src/main/java/io/grpc/okhttp/OkHttpServerTransport.java:299]: never mind about #9649.
I updated `server.shutdown()` to avoid forceful close.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**